### PR TITLE
Add Drag and drop component

### PR DESCRIPTION
**Pull request by Poltergeist**
This has been the work of @gargs0n

This adds the possibility to drop a JSON file onto the form / JSON editor and then uses the contents to populate the form and the JSON editor.

draganddropfiles mov

### CODE REVIEW

File: plugins/services/src/js/components/DragAndDrop.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  // the ondragenter and ondragover events
+  handleDragEvents(event) {
+    event.preventDefault();
+  }
+
+  handleOnDrop(event) {
+
+    event.preventDefault();
+
+    const files = event.dataTransfer ? event.dataTransfer.files : event.target.files;
```
Comment (mlunoe) [inline plugins/services/src/js/components/DragAndDrop.js]: > 80 chars

File: plugins/services/src/js/components/modals/NewCreateServiceModalForm.js [hunk 2]
```diff
@@ -184,6 +187,10 @@ class NewCreateServiceModalForm extends Component {
     });
   }
 
+  handleFileContents(appString) {
+    this.handleJSONChange(JSON.parse(appString));
```
Comment (mlunoe) [inline plugins/services/src/js/components/modals/NewCreateServiceModalForm.js]: We need a `try`-`catch` around this and a fallback if it fails, e.g. a modal

File: plugins/services/src/js/components/modals/NewCreateServiceModalForm.js [hunk 3]
```diff
@@ -258,11 +265,15 @@ class NewCreateServiceModalForm extends Component {
...
+      <DragAndDrop className="flex flex-item-grow-1"
+        handleFileContents={this.handleFileContents}>
+        <div
+          className="flex flex-item-grow-1 modal-body-offset gm-scrollbar-container-flex">
           <FluidGeminiScrollbar>
-            <div className="container flex flex-direction-top-to-bottom modal-body-padding-surrogate">
-              <form onChange={this.handleFormChange} onBlur={this.handleFormBlur}>
+            <div
+              className="container flex flex-direction-top-to-bottom modal-body-padding-surrogate">
+              <form onChange={this.handleFormChange}
```
Comment (mlunoe) [inline plugins/services/src/js/components/modals/NewCreateServiceModalForm.js]: #nit separate lines for the attributes

**Comment by mlunoe** [comment]
Also, we should probably have a modal to accept that you will overrode already entered data (if you have made a change).
How does this work, if you drag in a pod spec on the regular form and the other way around?
This doesn't seem to work with the JSON editor either

**Comment by wavesoft** [comment]
Ps. Might be a good idea to show a "Drop file here" message on hover. We can also show there a warning like, "dropping this file here will overwrite your existing data"

**Pull request closed without merging**